test/support/iterator_utils.h

```diff
@@ -240,6 +240,9 @@ struct iterator_traits_<T*>
 template <typename Iter, typename Tag>
 using is_same_iterator_category = ::std::is_same<typename iterator_traits_<Iter>::iterator_category, Tag>;
 
+template <typename Iter, typename Tag>
+using is_base_of_iterator_category = ::std::is_base_of<Tag, typename iterator_traits_<Iter>::iterator_category>;
```

Comment:
Current implementation is a bit confusing, e.g. one may think that  InputIterator is a base of random_access_iterator in the following expression:
```c++
is_base_of_iterator_category<InputIterator, ::std::random_access_iterator_tag>::value
```

Let's keep the original order of parameters to avoid confusion, i.e.:
```c++
template <typename Tag, typename Iter>
using is_base_of_iterator_category = ::std::is_base_of<Tag, typename iterator_traits_<Iter>::iterator_category>;
```

Reply:
This was done to make it easier to change is_same_iterator_category to is_base_of_iterator_category. But I agree that this can be misleading. And I will fix it.